File: 2nd year/MPP/project-mpp/csharp-project/persistence/validators/ResultValidator.cs
using model;

namespace persistence.validators
{
    public class ResultValidator : IValidator<Result>
    {
        public void Validate(Result result)
        {
            if (result == null)
            {
                throw new ValidationException("Entity not instance of result");
            }

            if (result.Value < 0)
            {
                throw new ValidationException("Result value can't be negative");
            }
        }
    }
}

File: 2nd year/MPP/project-mpp/java-project/TriathlonPersistence/src/main/java/triathlon/persistence/validators/ResultValidator.java
package triathlon.persistence.validators;

import triathlon.model.Result;

public class ResultValidator implements Validator<Result> {
    @Override
    public void validate(Result result) throws ValidationException {
        if (result == null) {
            throw new ValidationException("triathlon.model.entities.Entity not instance of result");
        }
        if (result.getValue() == null
                || result.getValue() < 0) {
            throw new ValidationException("Result value can't be negative");
        }
    }
}
